### DISCUSSION TITLE
Password reset feature in V2

### DISCUSSION THREAD

**Discussion by brentrobbins** [nhost/nhost#95] [Q&A] [answered]
Hello. I need to know how do you re-create the password reset feature in V2. I have it working well in V1, but I am not seeing this documented anywhere (yet).

**Comment by elitan** [comment, marked as answer]
This was fixed here https://github.com/nhost/hasura-auth/pull/58 and here https://github.com/nhost/hasura-auth/pull/47 but has not yet been released to Nhost users. We'll update everyone's apps and documentation as soon as possible. I'll write here in this thread once it's done.

**Reply by brentrobbins** [reply]
Thanks @elitan.  So it looks like this is for an authenticated user to update their password while logged in using: `nhost.auth.updateUser`? Is this correct and what are the params? Is it just the new password?

Also I was looking for the function to reset a user's password who is not logged in and can not remember their password. What's the function along with the params for that? Thanks for your help!

**Reply by elitan** [reply]
To update a user's password use:

```js
nhost.auth.changePassword({ newPassword });
```

Reference:

https://github.com/nhost/hasura-auth-js/blob/main/src/hasura-auth-client.ts#L357-L363
https://github.com/nhost/hasura-auth-js/blob/main/src/utils/types.ts#L104-L106

To reset a user's password use:

```js
nhost.auth.resetPassword({ email });
```

The `resetPassword`-method will send a link to the user's email, which will sign in the user. Then the user can use the same `nhost.auth.changePassword()` method.

We'll add more examples to this to make it more obvious how to use the different methods and how to set up the user flow.

**Reply by brentrobbins** [reply]
I am a bit confused since you mentioned both `nhost.auth.updatePassword` which seems to no longer exist and `nhost.auth.changePassword` comes back with an error that it requires the old password in the params (or at least that's the error I get when I try to just use the new password the params).

**Reply by elitan** [reply]
Sorry, should be `changedPassword`. I've edited my answer above.

The errors are probably because of this: https://github.com/nhost/nhost/discussions/95#discussioncomment-1897008

**Reply by brentrobbins** [reply]
Like I said I tried `nhost.auth.changePassword`, but I receive the following error:
`{message: 'Error validating request body. "oldPassword" is required.', status: 400}`

**Reply by brentrobbins** [reply]
The references above are for `@nhost/hasura-auth-js`, which I am not using. I am using `@nhost/react-auth`...is that the issue.

**Reply by elitan** [reply]
The reason you're getting the "oldPassword is required"-error is because we've not released the updates mentioned here: https://github.com/nhost/nhost/discussions/95#discussioncomment-1897008

`nhost.auth.changePassword` uses `@nhost/nhost-js` which imports `@nhost/hausra-auth-js` for everything under `nhost.auth.*`.

**Reply by brentrobbins** [reply]
OK good to know. Any eta when you'll deploy/merge this fix?

**Reply by elitan** [reply]
Maybe @nunopato can shed some light on the ETA?

**Reply by elitan** [reply]
This is now live for all Nhost apps.
